### DISCUSSION TITLE
Install it on jupyterhub

### DISCUSSION THREAD

**Discussion by amasanelli** [jupyterlab/jupyterlab-git#977] [Q&A] [answered]
Hello I'm running jupyterhub on a gcp VM. I followed these steps https://tljh.jupyter.org/en/latest/install/google.html to install jupyterhub

Now I'm trying to install jupyterlab-git but I can't successfully complete this task.

After I run the command and reload the browser an alert appear saying:

> Failed to load the jupyterlab-git server extension
> Git server extension is unavailable. Please ensure you have installed the JupyterLab Git server extension by running: pip install --upgrade jupyterlab-git. To confirm that the server extension is installed, run: jupyter server extension list.

`jupyter serverextension list`
> root@notebooks:~# jupyter server extension list
> Config dir: /home/jupyter-amasanelli/.jupyter
> 
> /opt/tljh/user/lib/python3.7/site-packages/jupyter_server/transutils.py:13: FutureWarning: The alias `_()` will be deprecated. Use `_i18n()` instead.
>   warnings.warn(warn_msg, FutureWarning)
> Config dir: /opt/tljh/user/etc/jupyter
>     jupyter_resource_usage enabled
>     - Validating jupyter_resource_usage...
>       jupyter_resource_usage  OK
>     jupyter_server_mathjax enabled
>     - Validating jupyter_server_mathjax...
>       jupyter_server_mathjax  OK
>     jupyterlab enabled
>     - Validating jupyterlab...
>       jupyterlab 3.0.16 OK
>     jupyterlab_git enabled
>     - Validating jupyterlab_git...
>       jupyterlab_git 0.30.1 OK
>     nbclassic enabled
>     - Validating nbclassic...
>       nbclassic  OK
>     nbdime enabled
>     - Validating nbdime...
>       nbdime 3.1.0 OK
>     voila.server_extension enabled
>     - Validating voila.server_extension...
>       voila.server_extension  OK
> 
> Config dir: /usr/local/etc/jupyter

Can yo help me?

**Comment by fcollonval** [comment]
Hey @amasanelli 

- how did you install the extension?
- could you please report the output of `jupyter serverextension list`?

**Reply by amasanelli** [reply]
Hi, I run the command `sudo pip install --upgrade jupyterlab-git` on
jupyterlab terminal

The output of `jupyter serverextension list` is on the body of my question

Thanks

**Reply by fcollonval** [reply]
There are two commands `jupyter serverextension list` and `jupyter server extension list` (note that one has a whitespace between `server` and `extension`) - the former is used by the classical notebook (and JupyterHub) when the second is used by `jupyter-server` and JupyterLab v3 (if not executed in JupyterHub context). In your first post, the terminal copy is showing `jupyter server extension list`. Hence my question to get the output of the other one `jupyter serverextension list`. But it may be safer to run again both commands to be sure.

**Reply by amasanelli** [reply]
Oh sorry about that.

Here is the output of both:

`jupyter serverextension list`
> config dir: /opt/tljh/user/etc/jupyter
>     jupyter_resource_usage  enabled 
>     - Validating...
>       jupyter_resource_usage  OK
>     jupyterlab  enabled 
>     - Validating...
>       jupyterlab 3.0.16 OK
>     jupyterlab_git  enabled 
>     - Validating...
>       jupyterlab_git 0.30.1 OK
>     nbdime  enabled 
>     - Validating...
>       nbdime 3.1.0 OK
>     nbgitpuller  enabled 
>     - Validating...
>       nbgitpuller 0.9.0 OK
>     nteract_on_jupyter  enabled 
>     - Validating...
>       nteract_on_jupyter 2.1.3 OK
>     voila.server_extension  enabled 
>     - Validating...
>       voila.server_extension  OK

`jupyter server extension list`
> Config dir: /home/jupyter-amasanelli/.jupyter
> 
> /opt/tljh/user/lib/python3.7/site-packages/jupyter_server/transutils.py:13: FutureWarning: The alias `_()` will be deprecated. Use `_i18n()` instead.
>   warnings.warn(warn_msg, FutureWarning)
> Config dir: /opt/tljh/user/etc/jupyter
>     jupyter_resource_usage enabled
>     - Validating jupyter_resource_usage...
>       jupyter_resource_usage  OK
>     jupyter_server_mathjax enabled
>     - Validating jupyter_server_mathjax...
>       jupyter_server_mathjax  OK
>     jupyterlab enabled
>     - Validating jupyterlab...
>       jupyterlab 3.0.16 OK
>     jupyterlab_git enabled
>     - Validating jupyterlab_git...
>       jupyterlab_git 0.30.1 OK
>     nbclassic enabled
>     - Validating nbclassic...
>       nbclassic  OK
>     nbdime enabled
>     - Validating nbdime...
>       nbdime 3.1.0 OK
>     voila.server_extension enabled
>     - Validating voila.server_extension...
>       voila.server_extension  OK
> 
> Config dir: /usr/local/etc/jupyter

**Comment by amasanelli** [comment, marked as answer]
I found the solution!

I installed it running `sudo pip install --upgrade jupyterlab-git` on the jupyterlab terminal and then I restarted the server using the jupyterhub control panel.

Now it's working!

Thanks

**Reply by fcollonval** [reply]
Great to hear

**Comment by fcollonval** [comment]
> Oh sorry about that.

Don't worry those small diff are hard to catch :wink:
